ledger/appcow.go

```diff
@@ -485,10 +485,18 @@ func (cb *roundCowState) StatefulEval(params logic.EvalParams, aidx basics.AppIn
 	}
 
 	// Eval the program
-	pass, err = logic.EvalStateful(program, params)
+	txnCost := 0
+	txnCost, pass, err = logic.EvalStateful(program, params)
 	if err != nil {
 		return false, basics.EvalDelta{}, ledgercore.LogicEvalError{Err: err}
 	}
+	if params.Proto.EnableAppFeePooling {
+		cb.pooledApplicationCosts += txnCost
```

Comment:
If I'm right about the problem where every app gets at least 700, I think this might catch the problem here.  So that's good.  But I don't think good enough.  We would like to catch the overage on the exact instruction that goes over, and stop.